src/shogun/machine/gp/SparseInference.cpp

```diff
@@ -99,16 +97,16 @@ SparseInference::SparseInference(std::shared_ptr<Kernel> kern, std::shared_ptr<F
 
 void SparseInference::init()
 {
-	SG_ADD(&m_inducing_features, "inducing_features", "inducing features",
-			ParameterProperties::HYPER | ParameterProperties::GRADIENT);
+	SG_ADD(
+	    &m_inducing_features, "inducing_features", "inducing features",
+	    ParameterProperties::MODEL | ParameterProperties::GRADIENT);
```

Comment:
actually these are hyper parameters. did we disuss this before and I said the opposite? :)

Reply:
yes, you said "These are not hyper parameters as they are learned. model parameter", should i change back?

Reply:
Would it make sense for it to be both? Because if the user doesn’t set it they will be inferred right?

Reply:
They are not learned during training but during model selection actually, so I was talking nonsense before... hyper! :)

Reply:
Oh I see, I thought inducing features were automagically chosen by the GPs. But I guess that is using this extension http://proceedings.mlr.press/v5/titsias09a/titsias09a.pdf

Reply:
This this method is implemented via gradient model selection when using Variational inference methods

Reply:
So yes

Reply:
so change back to HYPER?